lib/Stats.js

```diff
@@ -118,37 +151,225 @@ class Stats {
 			compilation.compiler.context === context
 				? compilation.requestShortener
 				: new RequestShortener(context);
-		const showPerformance = optionOrLocalFallback(options.performance, true);
-		const showHash = optionOrLocalFallback(options.hash, true);
-		const showEnv = optionOrLocalFallback(options.env, false);
-		const showVersion = optionOrLocalFallback(options.version, true);
-		const showTimings = optionOrLocalFallback(options.timings, true);
-		const showBuiltAt = optionOrLocalFallback(options.builtAt, true);
-		const showAssets = optionOrLocalFallback(options.assets, true);
-		const showEntrypoints = optionOrLocalFallback(options.entrypoints, true);
+
+		const showCachedAssets = isTypeOf(options.cachedAssets, "boolean")
+			? options.cachedAssets
+			: options.all !== undefined
+				? options.all
+				: assertTypeOf(
+						this.hooks.setDefault.for("cachedAssets").call(),
+						"boolean"
+					);
+
+		const showAssets = isTypeOf(options.assets, "boolean")
```

Comment:
From diff, looks like `isTypeOf` could be similar to old impl?

Reply:
With this implementation, I've added type checking too, we can't allow user to pass in any value of any type, if the value supplied by user if not of the type that we require, then we fallback to options.all ( for all boolean values, except for !toString values, will come to that later), if options.all is not set, then we take into account the default values ( which user has supplied either by their custom plugins or we've supplied in webpackOptionsApply), and type check them too, if they are also not of expected type, then we throw an error. However if expected type is not boolean, then we don't fallback to options.all, we directly assert value supplied by plugins.
For the values which are dependent on toString, we first check for current value, whether current value supplied by user is of supported type or not, if not, then we fallback to plugins, if no plugins are setting any value, then we can fallback to toString.

I think type checking the options value supplied should be done. However, if you insist, we can fallback to previous implementation too.